### PR TITLE
fix: i18n custom paths - SFS-3110

### PR DESCRIPTION
**Pull request by lemagnetic**
## What's the purpose of this pull request?

Stores configured with i18n custom paths (e.g. `https://store.com/europe/it`) returned **404** in production whenever the URL did not match the canonical locale code (e.g. `/it-IT`). The same URLs worked locally with `yarn dev`, but failed after `yarn build && yarn start` and on the standalone production deploy.

Three independent issues were colliding:

1. **`faststore start` was loading the wrong Next config.** `next start` was being spawned without the `.faststore` directory, so Next loaded a default config from the project root with `i18n = null`. Without `i18n`, the proxy threw `TypeError` on locale rewrites.
2. **Partytown was missing from production builds.** `partytown copylib` only ran via `predev`, so `/~partytown/partytown.js` was never copied into `public/` during `build` and the asset 404'd in production.
3. **Hostname validation was broken in standalone mode.** `request.nextUrl.hostname` resolves to the server bind address (`0.0.0.0` from the `HOSTNAME` env var set by the Dockerfile), not the client-facing host. The proxy compared this against binding hostnames (e.g. `brandless.fast.store`), so all binding rules were skipped and the catch-all returned 404.

## How it works?
**1. `packages/cli/src/commands/start.ts`** — pass `tmpDir` (the `.faststore` directory) as a positional argument to `next start`, so it loads the correct `next.config.js` with the `i18n` config derived from `discovery.config.js`. Using a positional arg (instead of changing `cwd`) keeps `next` resolvable from the project root via workspace hoisting.

```ts
return spawn(`${packageManager} next start ${tmpDir} -p ${port}`, { ... })
```

**2. `packages/cli/src/utils/generate.ts`** — mirror `predev` with a `prebuild` hook so Partytown is copied into `public/~partytown/` before the build, ensuring it ships in the standalone output.

```ts
predev: 'na run partytown',
prebuild: 'na run partytown',
```

**3. `packages/core/src/proxy.ts`** — read the hostname from the `Host` header instead of `request.nextUrl.hostname`, so binding lookup works consistently in both `next start` and standalone (`node server.js`) environments. `shouldValidateHostname` stays enabled.

```ts
const { pathname, search } = request.nextUrl
// In standalone, request.nextUrl.hostname is the server bind address
// (e.g. "0.0.0.0"), not the client-facing host. Use the Host header.
const hostname = request.headers.get('host')
```

## How to test it?

The PR is already deployed on `brandless.fast.store`, where both fixes can be verified:

- Open <https://brandless.fast.store/europe/it> → page renders correctly (no 404), confirming the custom path is resolved.
- DevTools → Network → `GET /~partytown/partytown.js` → **200 OK**, confirming Partytown is shipped with the build.
- Response of the page request includes the `x-middleware-rewrite` header pointing to the locale-prefixed path (e.g. `/it-IT/...`). 
image

- Canonical locale paths (`/it-IT`, `/en-GB`, `/pt-BR`) keep returning 200 — no regression.

## References

- Jira: [SFS-3110 — i18n custom paths](https://vtex-dev.atlassian.net/jira/software/c/projects/SFS/boards/1051?selectedIssue=SFS-3110)

## Release Notes

* **Refactor**
  * Centralized hostname extraction logic across API endpoints for improved consistency and reliability.

* **Tests**
  * Added comprehensive test suite for hostname normalization utility covering edge cases and fallback scenarios.

* **Chores**
  * Added partytown prebuild script to the build process.
  * Improved CLI start command execution with better process handling.

### CODE REVIEW

File: packages/core/src/proxy.ts [hunk 1]
```diff
@@ -97,7 +97,17 @@ export function proxy(request: NextRequest) {
...
+
+  // Read the hostname from the Host header rather than request.nextUrl.hostname.
+  // When the Next.js standalone server is started with a `hostname` (which the
+  // generated server.js always does, defaulting to "0.0.0.0" via the HOSTNAME
+  // env var), Next.js builds the internal request URL using that bind address
+  // instead of the client-facing host, see attachRequestMeta in
+  // next/dist/server/next-server.js. As a result, request.nextUrl.hostname
+  // returns the bind address (e.g. "0.0.0.0"), which never matches the
+  // hostname extracted from a binding URL (e.g. "brandless.fast.store").
+  const hostname = request.headers.get('host')
```
[the remark this answers is not in the document]
Comment (gabpaladino) [reply packages/core/src/proxy.ts]: acho justo essa alteração, lembro que em outra parte do grapqhl o Formiga tinha feito esse tipo de proteção, não da pra reaproveitar logica?

File: packages/core/src/proxy.ts [hunk 2]
```diff
@@ -97,7 +98,17 @@ export function proxy(request: NextRequest) {
...
+  const { pathname, search } = request.nextUrl
+
+  // Read the hostname from the Host header rather than request.nextUrl.hostname.
+  // When the Next.js standalone server is started with a `hostname` (which the
+  // generated server.js always does, defaulting to "0.0.0.0" via the HOSTNAME
+  // env var), Next.js builds the internal request URL using that bind address
+  // instead of the client-facing host, see attachRequestMeta in
+  // next/dist/server/next-server.js. As a result, request.nextUrl.hostname
+  // returns the bind address (e.g. "0.0.0.0"), which never matches the
+  // hostname extracted from a binding URL (e.g. "brandless.fast.store").
```
Comment (eduardoformiga) [inline packages/core/src/proxy.ts:110]: 👀

File: packages/core/test/utils/getRequestHostname.test.ts
Comment (eduardoformiga) [inline packages/core/test/utils/getRequestHostname.test.ts:1]: 💯

**Review by eduardoformiga — APPROVED** [review]
The changes LGTM!
I followed the steps to test, and it worked as mentioned 😉 .
